internal/pkg/service/buffer/file/file_create.go

```diff
@@ -0,0 +1,72 @@
+package file
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/keboola/go-client/pkg/keboola"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/rollback"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+func (m *Manager) CreateFilesForReceiver(ctx context.Context, rb rollback.Builder, receiver *model.Receiver) error {
+	rb = rb.AddParallel()
+	wg := &sync.WaitGroup{}
+	errs := errors.NewMultiError()
+
+	for i := range receiver.Exports {
+		export := &receiver.Exports[i]
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			if err := m.WithToken(export.Token.Token).CreateFileForExport(ctx, rb, export); err != nil {
+				errs.Append(err)
+			}
+		}()
+	}
+
+	wg.Wait()
+	return errs.ErrorOrNil()
+}
+
+func (m *AuthorizedManager) CreateFileForExport(ctx context.Context, rb rollback.Builder, export *model.Export) error {
```

Comment:
`file.Manager` ma metodu `WithToken`, ktora vrati `AuthorizedManager`, ktory mas uz konkretne operacie, ktore potrebuju token.

Je to takto riesene primarne pre Worker, kde sa vacsinou iteruju vsetky exporty, ktory ma worker pridelene, a kazdy ma svoj vlastny token, ktorym robi operacie. Teda nemoze to byt nejaka jedna zavislost, ktora sa vsade pouziva, ale pred vykonanim operacie si musis vytvorit `AuthorizedManager`.

Predtym bol okolo toho v kode chaos, na roznych miestach sa vytvarala autorizovana API instancia.